### ISSUE TITLE
PAM service configuration blocks `login` for AD users

### ISSUE THREAD

**Issue by wucke13** [NixOS/nixpkgs#128116]
**Describe the bug**
When using krb5 together with samba & winbind for login in an AD, login works everywhere except `login`. So `su` or `sshd` for example work just fine. However, when `login` is used (either through the binary in the shell or through a tty or display manager), I get:

```
Password:
You have been logged on using cached credentials.

Login incorrect
```

Up on further inspection, I noticed a difference in the respective pam service files. Most of these files have 

```pam
auth sufficient pam_unix.so   likeauth try_first_pass
```
with one notable difference: `/etc/pam.d/login` has

```pam
auth required pam_unix.so nullok  likeauth
```
If I change the `required` in `/etc/pam.d/login` by `sufficient`, I can login to AD accounts without problem even using `login` or `tty`.

**To Reproduce**
(It's not realistically possible to reproduce the full setup)
1. Setup an AD with krb5, samba (+winbind) and a custom pam winbind service
````nix
security.pam.services = {
  login.makeHomeDir = true;
  sshd.makeHomeDir = true;
  su.makeHomeDir = true;
  winbind.text = ''
    auth sufficient ${pkgs.samba4Full}/lib/security/pam_winbind.so
    account sufficient ${pkgs.samba4Full}/lib/security/pam_winbind.so
    password sufficient ${pkgs.samba4Full}/lib/security/pam_winbind.so
    session sufficient ${pkgs.samba4Full}/lib/security/pam_winbind.so
  '';
};
````

**Expected behavior**
Login works for AD users with `su`, `sshd` and `login` all the same.

**Additional Notes**

I'm by no means an expert in the domain of _domains_. What I however wonder is, how others solved this issue when integrating NixOS machines in an AD.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.11, NixOS, 21.05.20210620.6613a30 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-21.05.1076.bad3ccd099e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by wucke13** [comment]
Update: This seems to be related to the section folloowing [this line](https://github.com/NixOS/nixpkgs/blob/d1aa855491b03eeb91a474deb43df89a7c03f484/nixos/modules/security/pam.nix#L420) in the pam module.

Namely, if any of `config.security.pam.enableEcryptfs`, `cfg.pamMount`, `cfg.enableKwallet`, `cfg.enableGnomeKeyring`, `cfg.googleAuthenticator.enable`, `cfg.gnupg.enable` or `cfg.duoSecurity.enable` is true, `pam_unix.so` is `required`, breaking any remote auth (like LDAP/AD/...). Why is this the case? Do these things (like kwallet or gnome-keyring) only work when called after `pam_unix.so`?

**Comment by wucke13** [comment]
At least for sssd, it looks like gnome keyring would be propagated correctly ([sssd 2.2.2 release notes](https://sssd.io/release-notes/sssd-2.2.2.html)). So probably the comment

>  Modules in this block require having the password set in PAM_AUTHTOK.
    pam_unix is marked as 'sufficient' on NixOS which means nothing will run
    after it succeeds. Certain modules need to run after pam_unix
    prompts the user for password so we run it once with 'required' at an
    earlier point and it will run again with 'sufficient' further down.
    We use try_first_pass the second time to avoid prompting password twice
    
from https://github.com/NixOS/nixpkgs/blob/d1aa855491b03eeb91a474deb43df89a7c03f484/nixos/modules/security/pam.nix#L413 is likely not true anymore for sssd

**Comment by fpletz** [comment]
Fixed by #173495.
